### ISSUE TITLE
java.lang.BootstrapMethodError: too many bootstrap method arguments

### ISSUE THREAD

**Issue by nevillelyh** [spotify/scio#468]
This happens on Scala 2.12, from `MultiJoin.scala`:
```
[info]   Cause: java.lang.BootstrapMethodError: too many bootstrap method arguments
[info]   at java.lang.invoke.CallSite.makeSite(CallSite.java:320)
[info]   at java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:307)
[info]   at java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:297)
[info]   at com.spotify.scio.util.MultiJoin.$deserializeLambda$(MultiJoin.scala)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   at java.lang.invoke.SerializedLambda.readResolve(SerializedLambda.java:230)
[info]   at sun.reflect.GeneratedMethodAccessor23.invoke(Unknown Source)
```

Reducing the multi-join arity fixes the problem.

**Comment by retronym** [comment]
Tracked as https://issues.scala-lang.org/browse/SI-10232

**Comment by nevillelyh** [comment]
Verified that this is fixed by Scala 2.12.3
